### PR TITLE
add knockback3 package

### PR DESCRIPTION
**Pull request by Cokemonkey11**
See https://github.com/peq/WurstScript/pull/472 for history and discussion

### CODE REVIEW

**Comment by Frotty** [comment]
> [00:22]   @Frotty: the system should not hardcoded destroy destructables, add some short description at the top if you want this to be done with 0 ENUM_SIZE, or offer a listener, code var, whatever

Also you could add configuration to only destroy trees and not all destructables. But then again, offering a way to add a listener would let the user do this instead.

**Comment by Cokemonkey11** [comment]
@Frotty regarding tree checking, Knockback3D for vJass supports it, but it uses `IsDestructableTree` library which we don't have so I removed it.

http://www.wc3c.net/showthread.php?t=103927

**Comment by Cokemonkey11** [comment]
@Frotty agree that adding proper listener would be ideal but I think there's a lot of code shared with colliders. I think this should be a separate topic.

This should be ready now.

File: wurst/systems/Knockback3.wurst [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    */
+    @configurable static constant USE_PROP_WINDOW_MODIFIERS = true
+
+
+    /**
+    Apply a knockback vector to unit u. `velocity` is initial speed in units
+    per second. `groundAngle` and `airAngle` are the direction and trajectory
+    parameters, respectively. Example:
+
+        import Knocback3
```
Comment (Frotty) [inline wurst/systems/Knockback3.wurst]: typo

File: wurst/systems/Knockback3.wurst [hunk 2]
```diff
@@ -0,0 +1,225 @@
+/** By default, this package allows units to collide with destructables and
+    destroy them. To disable this feature, make the global
+    destroyDestructableSpeedThreshold a very high value:
+
+       Knockback3.destroyDestructableSpeedThreshold = 999999999.
```
Comment (Frotty) [inline wurst/systems/Knockback3.wurst:5]: Then it needs to be configurable (with dependency management u wont be able to modify stdlib)

**Comment by Frotty** [comment]
I think what many people want from a 3d knockback is also a jump like function, ie being able to specify a target position and the knockback will be set so that the unit lands on the target pos. please provide an overloaded add function with different parameters.
Otherwise looks gut so far 🍰

File: wurst/systems/Knockback3.wurst (hunk 2 above)
Comment (Cokemonkey11) [inline wurst/systems/Knockback3.wurst:5]: The value is `var` not `constant`, so my example is meant to show how to change it from the context of some function that imports Knockback3, not suggesting modifying it.

Do you still prefer constant with `@configurable`? If so, is that just convention?

**Comment by Cokemonkey11** [comment]
@Frotty done :) and there's a cute demo in the demo map too

File: wurst/systems/Knockback3.wurst (hunk 2 above)
Comment (Frotty) [inline wurst/systems/Knockback3.wurst:5]: No but why would this value change ingame

Comment (Cokemonkey11) [inline wurst/systems/Knockback3.wurst:5]: In microrunnertd there are high gravity rounds, for example.

**Comment by Frotty** [comment]
🌴 💃 👙

**Pull request merged**